src/PerunDummy.hs

```diff
@@ -298,26 +349,28 @@ mkChannelValidator cID oldDatum action ctx =
     getPOSIXTimeFromUpperBound (UpperBound (Finite t) _) = t
     getPOSIXTimeFromUpperBound _ = traceError "unable to verify time"
 
+    -- | Check that the consuming transaction is valid in an interval of at most `defaultValidMsRange` milliseconds
     allowedValidRangeSize :: Bool
-    allowedValidRangeSize = (getPOSIXEndTime (strictUpperBound (txInfoValidRange info)) - getPOSIXStartTime (strictLowerBound (txInfoValidRange info))) <= defaultValidMsRange
+    allowedValidRangeSize = getPOSIXEndTime (strictUpperBound (txInfoValidRange info)) - getPOSIXStartTime (strictLowerBound (txInfoValidRange info)) <= defaultValidMsRange
 
-    -- Check that the time in the output datum of the consuming transaction is located inside the valid range of the consuming transaction
+    -- | Check that the time in the output datum of the consuming transaction is located inside the valid range of the consuming transaction
     outputTimeInValidRange :: Bool
     outputTimeInValidRange = Ledger.member (time outputDatum) (txInfoValidRange info)
 
-    -- Check that the relative time-lock has passed
+    -- | Check that the relative time-lock has passed
     correctForceCloseSlotRange :: Bool
     correctForceCloseSlotRange = from (time oldDatum + fromMilliSeconds (DiffMilliSeconds (pTimeLock (channelParameters oldDatum)))) `contains` txInfoValidRange info
 
     -- Returns true if party h is payed value v in an output of the transaction
-    getsValue :: PaymentPubKeyHash -> Value -> Bool
-    getsValue pkh v =
+    getsValue :: PaymentPubKeyHash -> Integer -> Bool
+    getsValue pkh v = (v == 0) || (
```

Comment:
NVM, my mind is blown and I know why after encountering the error message myself. My proposal breaks plutus compilation and rightfully so.

Using my way lets `GHC` use default Haskell desugaring, resulting in something along the lines of:

```haskell
getsValue pkh v = if v Haskell.Prelude.(==) 0 then ... else ...
```
Of course this breaks Plutus because `==` is not inlineable :bulb: